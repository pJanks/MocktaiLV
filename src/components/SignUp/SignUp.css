section {
  display: flex;
}

form {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}
