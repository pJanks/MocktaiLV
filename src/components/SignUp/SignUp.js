import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './SignUp.css'

class SignUp extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      password: ''
    }
  }

  handleChange = async (event) => {
    await this.setState({ [event.target.name]: event.target.value });
  }

  handleSignUpButtonClick = (event, name, password) => {
    event.preventDefault()
    localStorage.setItem(name, JSON.stringify(this.state));
  }


  render() {
    return (
      <section>
      <form>
        <input
          type="text"
          name="name"
          placeholder="Name"
          onChange={this.handleChange}
        />
        <input
          type="password"
          name="password"
          placeholder="Password"
          onChange={this.handleChange}
        />
        <button onClick={(event) => this.handleSignUpButtonClick(event, this.state.name, this.state.password)}><Link className='link' to='/'>Sign Up</Link></button>
      </form>
      </section>
    )
  }
}

export default SignUp
