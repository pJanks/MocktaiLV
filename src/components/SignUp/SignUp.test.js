import React from 'react';
import SignUp from './SignUp';
import { shallow } from 'enzyme'

describe('SignUp', () => {


  it('should match a snapshot as expected', () => {
    const wrapper = shallow(<SignUp />)

    expect(wrapper).toMatchSnapshot()
  })
})
