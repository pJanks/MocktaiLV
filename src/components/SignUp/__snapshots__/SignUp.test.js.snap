// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match a snapshot as expected 1`] = `
<section>
  <form>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <button
      onClick={[Function]}
    >
      <Link
        className="link"
        to="/"
      >
        Sign Up
      </Link>
    </button>
  </form>
</section>
`;
